sklearn/neighbors/_classification.py

```diff
@@ -229,6 +230,19 @@ def predict(self, X):
             Class labels for each data sample.
         """
         if self.weights == "uniform":
+            if self.algorithm == "brute" and ArgKminLabels.is_usable_for(
+                X, self._fit_X, self.metric
+            ):
+                probabilities = self.predict_proba(X)
+                if self.outputs_2d_:
+                    return np.stack(
+                        [
+                            self.classes_[idx][np.argmax(probas, axis=1)]
+                            for idx, probas in enumerate(probabilities)
+                        ],
+                        axis=1,
+                    )
+                return self.classes_[np.argmax(probabilities, axis=1)]
```

Comment:
Not a requirement to get this PR merged, but I would be very interested in a profile report for this Python snippet relative to a global time it takes to call `predict` on a dataset with 1m+ training datapoints and a few hundreds prediction points on a machine with 16 physical cores or more.

Reply:
I checked on the following case (with a large number of classes, large number of neighbors, a rather smallish number of features and a test set as large as the training set)

```python
from sklearn.neighbors import KNeighborsClassifier
from joblib import Memory
from sklearn.model_selection import train_test_split
import sys
import time


@Memory(location="/tmp/joblib").cache
def make_bench_data(n_samples):
    from sklearn.datasets import make_classification

    X, y = make_classification(
        n_samples=n_samples,
        n_features=30,
        n_classes=100,
        n_clusters_per_class=2,
        n_informative=25,
        random_state=0,
    )

    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size=0.5, random_state=0
    )
    return X_train, X_test, y_train, y_test


X_train, X_test, y_train, y_test = make_bench_data(10000)
clf = KNeighborsClassifier(algorithm="brute", n_neighbors=500).fit(X_train, y_train)

n_repeats = int(sys.argv[1]) if len(sys.argv) > 1 else 1

tic = time.perf_counter()
for i in range(n_repeats):
    clf.predict(X_test)
print(f"Time: {time.perf_counter() - tic:.3f} s")
```

and this seems to be negligible, the extra overhead of the tail of the `predict` call (line 224) is barely visible (on the right) compared to the `predict_proba` call (line 285):

<img width="1052" alt="image" src="https://user-images.githubusercontent.com/89061/217618269-6ce4b5e6-2306-4299-8cc0-ca18a3c0f90f.png">
w

Reply:
Actually, I am confused, this does not call the new kernel but goes through kneighbors and then does the predict proba normalization. I need to investigate why it does not call the new Cython code.

Reply:
Hum probably because this code is disabled for euclidean / sqeuclidean. I need to redo profiling with Manhattan instead. Will continue tomorrow.

Reply:
Thanks. Not that it's quite confusing to use the ".svg" extension for the json report. For information, the speedscope / json formatted reports have to be open with the speedscope.app viewer and cannot directly be open in a browser as the flamegraph / svg reports.

Reply:
I tried to add the `# cython: profile=True` comment to the 
`_argkminclassmode.pyx.tp` source file to get viztracer to profile the Cython part but I cannot get it work after many tenatives (I do not see the Cython calls in my viztracer reports), even in monothreaded mode.

Have you managed to ever get it to work on your side? Is there a central way to enable the Cython profile flag in the scikit-learn build config?

Unfortunately I cannot use py-spy with the `--native` flag as it's only implemented for linux x86_64.